File: task2.js
const arr = [1,2,3,4,5,6,7,8,9,10,11, 12, 13, 14, 15, 16]
const x = 5

const binarySearch = (arr, x) => {
    let lowIndex = 0
    let highIndex = arr.length - 1
    let mid = 0
    let count = 0

    while (lowIndex <= highIndex) {
         count++

        mid = Math.floor((lowIndex + highIndex) / 2) 
        console.log("mid index:", mid, "item:", arr[mid]);
        
        if (arr[mid] === x) {
            console.log("count", count);
            
            return arr[mid]
        }

        if (arr[mid] > x) {
            highIndex = mid - 1
        } else {
            lowIndex = mid + 1     
        }
       
    }
    return -1 


}

console.log(binarySearch(arr, 8));
 

File: task2.py
arr = [1,2,3,4,5,6,7,8,9,10,11, 12, 13, 14, 15, 16]

def binary_search(arr, x):
    low = 0
    high = len(arr) - 1
    result = None
    count = 0 

    while (low <= high):
        count += 1
        mid = (high + low) // 2

        if arr[mid] >= x:
            result = arr[mid]
            high = mid - 1
        else:
            low = mid + 1

  
    return (count, result)


print(binary_search(arr, 5))